spec/input_handler_spec.rb

```diff
@@ -0,0 +1,6 @@
+# Not sure how to test input handling in an RSpec test
```

Comment:
This is sort of a boundary of your application, so I think something like testing the interactions happen as expected. Sort of like what we did with `Mock<IMyDependency>` in C#, you can look into some similar test doubles for ruby here: https://relishapp.com/rspec/rspec-mocks/docs